### PR TITLE
 add error pages

### PR DESCRIPTION
**Pull request by marwa7med**

### CODE REVIEW

**Comment by marwa7med** [comment]
#32

File: src/controllers/error.js [hunk 1]
```diff
@@ -1 +1,17 @@
+const path = require('path');
```
Comment (mhmdshorafa) [inline src/controllers/error.js:1]: we do not use path in this file

Comment (marwa7med) [reply src/controllers/error.js:1]: this path for 404 error page

Comment (mhmdshorafa) [reply src/controllers/error.js:1]: yeah ,, we required path ,, but it's actually not used yet we can add it when we use it  (y)

Comment (marwa7med) [reply src/controllers/error.js:1]: ok I just set it up to start working in this pages :)

**Review by Alaa-Khattab — CHANGES_REQUESTED** [review]
* Please, choose suitable name for your branch such as : `error-pages` or `handle-error` or any other  name have a meaning  
* Fix the test because 7 pass and 6 fail

File: package.json [hunk 2]
```diff
@@ -6,7 +6,11 @@
   "scripts": {
     "start": "node src",
     "start:watch": "nodemon src/index.js",
+
+    "test:router": "node src/test/testRouter.js"
```
Comment (Alaa-Khattab) [inline package.json:10]: missing a comma in line 10

File: src/app.js [hunk 3]
```diff
@@ -22,6 +22,9 @@ app.engine('hbs', hbars({
 }));
 
 app.set('port', process.env.PORT || 4000);
+
+app.use(favicon(path.join(__dirname, '..', 'public', 'images/icon.ico')));
```
Comment (Alaa-Khattab) [inline src/app.js:26]: favicon not defined because you did not require it !!

File: src/controllers/index.js [hunk 4]
```diff
@@ -2,8 +2,15 @@ const express = require('express');
...
+const error = require('./error');
 const levels = require('./levels');
 
-router.get('/', home.get);
-router.get('/levels', levels.get);
-module.exports = router;
+module.exports = router.get('/', home.get)
+  .get('/contact', contact.get)
+  .get('/about', about.get)
+  .get('error404', error.get)
```
Comment (Alaa-Khattab) [inline src/controllers/index.js:14]: change the line 14 into `.get('/error404', error.client)` in order to make it work

**Review by Alaa-Khattab — CHANGES_REQUESTED** [review]
what is the difference between this pull request and this one #33 ?

**Pull request closed without merging**